src/custom-error.ts

```diff
@@ -0,0 +1,21 @@
+import { GraphQLError } from 'graphql';
+
+export class CustomError extends Error {
+  code: number;
+  additionalInfo?: string;
+
+  constructor(code: number, message: string, additionalInfo?: string) {
+    super(message);
+    this.code = code;
+    this.additionalInfo = additionalInfo;
+  }
+}
+
+export const formatError = (error: GraphQLError) => {
+  const formattedError = error.originalError as CustomError;
+  return {
+    code: formattedError.code,
+    message: formattedError.message,
+    additionalInfo: formattedError?.additionalInfo,
+  };
```

Comment:
O que acontece se o erro jogado não for um CustomError? Pode acontecer de outra biblioteca jogar erro, ou até algum erro interno no código.

Reply:
Nesse caso, seria melhor retornar o erro lançado e suas propriedades direto do GraphQL, ou algum tipo de erro genérico 'Unknown Error' para manter a consistência de code - message - additionalInfo?

Reply:
Pensei em algo do tipo: se não for CustomError, retornar code 500, 'Erro interno'.